### DISCUSSION TITLE
Audio quality on WiiM mini devices in MA audio group

### DISCUSSION THREAD

**Discussion by JimmyMPage** [music-assistant/support#1369] [Q&A] [answered]
Tonight I added a fifth WiiM to my audio group and we were enjoying some music when my son (bless his young audiophile heart) noticed the music sounded slightly distorted. Sure enough, he was right.

I don't think this was an issue before I added drhe 5th WiiM Mini. The provider is Tidal and the track showed FLAC in MA. All players have compression and normalization turned off.

When playing to the WiiMs as a group from the WiiM app the audio is noticeably higher quality.

Is this just the nature of MA's way of playing to the group or is this something I can fix? Network bandwidth shouldn't be an issue, but I suppose it's possible there is some saturation or bottlenecking going on (the router is connected to the network via a LAGG with 3 1ghz NICs and all WiiMs are on 5ghz wireless with solid connections).

I enjoy troubleshooting this kind of thing, but I don't have a direction to start working.

**Comment by marcelveldt** [comment]
What kind of protocol is used here ? Airplay ?

1) You really should not send PCM to the players, especially not to 5 players at once. Your network will get congested in a blip. Just use FLAC, its lossless and we've set the compression level to 0 so I can ensure that is lossless. I should perhaps add a big fat warning here.

2) Normalization is also fine to use as I'm using a method that does not impact dynamic range or quality by just adjusting the gain based on the EVU-R128 measurement. It might even improve dynamic range because of the slightly reduced gain, leaving more headroom. Especially with streaming providers nowadays with their loadness war to encode everything at max level.

Reading your story this feels like network congestion to me. Some players will just skip frames if they do not come fast enough, some others will just halt playback until a certain buffer is full.

Can you star playing to the players one by one to start with ?
And let me know what kind of protocol and group you are using ?

**Reply by JimmyMPage** [reply]
I am using Airplay protocol. I have re-enabled compression on all players and the group. I've also re-enabled normalization. When I play the players one at a time, there was a significant change in audio quality and it compared favorably to playing each player through the native (WiiM) app. I did NOT start them one at a time and work up to 5. I will do that and report back. I also think it was a network issue...I haven't had any congestion issues since I went to a managed switch and a LAGG connection to my OPNSense router, but I was NOT monitoring the router when this was occurring, and it strikes me that I could easily do that. Thanks for the pointers!

**Reply by JimmyMPage** [reply]
OK. I tested this morning and found the issue is present ONLY when playing with the group. Even with only one of the 5 players "turned on" and playing, the distortion was present. When using that same player, with the same protocol (Airplay) NOT included in the group, there was no distortion or artifacts (it sounds fantastic).

This seems like a gain issue. Somewhere in the signal chain there is clipping (which is what the distortion has sounded like to my ears from the beginning...but I didn't think that was possible). The volume for each player is less than 50%, and the overall group volume seems linked to the sum of each player's volume, so I'm not sure how I could be overdriving anything. The WiiMs are not overdriving the input on the amps as the problem goes away when playing from the WiiM app or when playing from MA with the individual player selected. In addition, there is no audible change in VOLUME when switching between the individual player in MA and when playing through each player by selecting them within the audio group...the distortion is there when playing each player via the group...and gone when playing each player by selecting them individually outside the group.

**Reply by marcelveldt** [reply, marked as answer]
What level did you set the volume normalization of the group player ? My recommendation would be to set it at -17 LUFS

**Comment by JimmyMPage** [comment]
That did it! I had it set at zero for some reason. I don't really understand that setting, but I am not sure why I fiddled with it...dumb. Anyway...that was a simple fix! Thanks for the help.

**Reply by marcelveldt** [reply]
ah zero is leaving no headroom, that would cause clipping big time. I believe the official EBU recommendation is even -22 but that might sound a bit low. -17 leaves plenty of headroom and no chance of clipping.

**Reply by JimmyMPage** [reply]
Perfect. Thanks so much Marcel!

**Reply by marcelveldt** [reply]
Enjoy your music and I'll let you know if there's any progress on the linkplay support.
